### PR TITLE
feat(code view): code view navigation from diagram

### PR DESCRIPTION
**Pull request by vivjd**
## Overview
Completes #97 

## Proposed Changes
- Click navigation to code view directly when the user clicks on a component. Refactored mock data to reflect this change.
- Updated file explorer sidebar: added text truncation for file name, aligned nested layers in a more concise fashion

<details>
<summary>Screenshots of your changes (if applicable)</summary>

Clickable Components to Code View:
https://github.com/user-attachments/assets/1bd9586b-0624-4dea-871d-c7058df64023

Enhanced Side Bar:
https://github.com/user-attachments/assets/ed6d84ae-5c5d-4b77-a314-81c5bed90286

</details>

## How to Test & Review

_(Help your reviewer by providing clear instructions on how to verify your changes. Include specific commands, edge cases to watch out for, or UI navigation paths.)_

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |    x      |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |    x      |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |          |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |          |
| 📚 _Documentation update_ (change that _only_ updates documentation)                    |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

- [x] I have performed a self-review of my changes.
    - Check that all changed files included in this pull request are intentional changes.
    - Check that all changes are relevant to the purpose of this pull request, as described above.
- [x] I have added tests for my changes, if applicable.
    - This is **required** for all bug fixes and new features.
- [ ] I have updated the project documentation, if applicable.
    - This is **required** for new features.
- [x] **(Frontend)** I have added/updated text in `i18n` JSON files for any new user-facing strings to support multilingual features.
After opening your pull request:

- [x] I have verified that the CI tests have passed.
- [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer, and a fellow student.
- [ ] Technical Debt: If temporary workarounds or "TODOs" were used, I have opened a tracking issue to address them properly.
    - Linked Issues: 

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

### CODE REVIEW

**Review by missionStrawberry — CHANGES_REQUESTED** [review]
Two things: there's no consistency in the open status of the side bar between the code and diagram views. It's not super annoying, but the way the page is built suggests it should have continuity. 
The second thing is that nodes have the hover state when they're not actually clickable. This issue could disappear when we're not using the mock data.

**Comment by vivjd** [comment]
> Two things: there's no consistency in the open status of the side bar between the code and diagram views. It's not super annoying, but the way the page is built suggests it should have continuity. The second thing is that nodes have the hover state when they're not actually clickable. This issue could disappear when we're not using the mock data.

The side bar issue should be fixed now! Also added better mock data so that we show multiple violations.

The hover state should be fine once we switch to actual data.

File: clean-architecture-visualizer/frontend/src/mocks/data/analysis.ts [hunk 1]
```diff
@@ -150,10 +157,80 @@ export const mockInteractionDetails = {
...
+      target: "UserSignOutDataAccessInterface",
+      type: "INHERITANCE",
       status: "VIOLATION"
+    },
+    {
+      id: "edge-11",
+      source: "UserSignOutInteractor",
+      target: "Database",
+      type: "DEPENDENCY",
+      status: "INCORRECT_DEPENDENCY"
```
Comment (missionStrawberry) [inline clean-architecture-visualizer/frontend/src/mocks/data/analysis.ts:219]: You need to add these status types to the i18n file mapping that gives the title of the violation. I think I only added a mapping for incorrect dependency.

Comment (missionStrawberry) [reply clean-architecture-visualizer/frontend/src/mocks/data/analysis.ts:219]: for the mockViolations I mean

**Review by missionStrawberry — CHANGES_REQUESTED** [review]
The sidebar having a consistent open state is working great! Thanks for fixing that. 
For the violations, there are some translations missing in the i18n folder since I only added one initially. 
It would also be good to have mock data for a use case without violations if it shows there are no violations on the checker mode page, but learning mode is showing the ideal case right now so feel free to leave that as a ticket for future development. Same thing with having more than one case, although I like that this set of mock data is more extensive.

**Review by missionStrawberry — APPROVED** [review]
I fixed the hover state being on for missing nodes, and found the edge that shouldn't have been there. All looks good now!

**Review by matt-dahlgren — APPROVED** [review]
